src/map-context.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Map } from 'mapbox-gl';
+export interface MapContext {
+  map: Map;
+}
+export type Props = Partial<MapContext>;
+// tslint:disable:no-object-literal-type-assertion
```

Comment:
`map` is required in `MapContext` but it's actually not always there, with the default value of createContext here... this could *potentially* create problems, right?

Reply:
Yes and no, the same situation was the case before with the old context and as before the children are not mounted in the tree until the map is `ready` (see `Map.componentDidMount` & `Map.render`) so unless the `ready` state can be `true` without being set in `map.on("load", ...)` then that could potentially create problems? But I cant really see how, unless I am missing something?